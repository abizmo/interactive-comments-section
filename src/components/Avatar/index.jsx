import React from 'react';
import PropTypes from 'prop-types';

function Avatar({ user }) {
  return (
    <picture style={{ aspectRatio: 1, width: '2rem' }}>
      <source srcSet={`${process.env.PUBLIC_URL}/${user.image.webp}`} type="image/webp" />
      <img style={{ width: '100%' }} src={`${process.env.PUBLIC_URL}/${user.image.png}`} alt={user.username} />
    </picture>
  );
}

Avatar.propTypes = {
  user: PropTypes.shape({
    image: PropTypes.shape().isRequired,
    username: PropTypes.string.isRequired,
  }).isRequired,
};

export default Avatar;
