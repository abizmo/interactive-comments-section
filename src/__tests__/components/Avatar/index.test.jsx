import React from 'react';

import { render, screen } from '@testing-library/react';
import Avatar from '../../../components/Avatar';

const user = {
  image: {
    png: '../../../assets/images/avatars/image-juliusomo.png',
    webp: '../../../assets/images/avatars/image-juliusomo.webp',
  },
  username: 'juliusomo',
};

test('should render', () => {
  const { container } = render(<Avatar user={user} />);

  expect(screen.queryByAltText(user.username)).toBeTruthy();
  expect(container).toMatchSnapshot();
});
