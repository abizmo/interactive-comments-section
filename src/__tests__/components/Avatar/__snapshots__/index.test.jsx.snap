// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <picture
    style="width: 2rem;"
  >
    <source
      srcset="/avatars/amyrobson.webp"
      type="image/webp"
    />
    <img
      alt="amyrobson"
      src="/avatars/amyrobson.png"
      style="width: 100%;"
    />
  </picture>
</div>
`;
